test/Kestrel.InMemory.FunctionalTests/Http2/Http2StreamTests.cs

```diff
@@ -836,18 +836,19 @@ public async Task ContentLength_Received_MultipleDataFramesOverSize_Reset()
             await StartStreamAsync(1, headers, endStream: false);
             await SendDataAsync(1, new byte[1].AsSpan(), endStream: false);
             await SendDataAsync(1, new byte[2].AsSpan(), endStream: false);
+            // This will cause the content length to go bad
             await SendDataAsync(1, new byte[10].AsSpan(), endStream: false);
-            await SendDataAsync(1, new byte[2].AsSpan(), endStream: true);
```

Comment:
It's a product bug if we can't handle this race, not a test bug. The frame is likely already in flight when we send the RST and we need to handle it.

Reply:
Can you elaborate? It's not a race the test is just broken. If you send to an aborted stream the entire connection is aborted. Is that not the expected behavior?

Reply:
No that's not acceptable behavior for the server. The client can have multiple frames in flight before the stream is aborted. Those frames need to be filtered out rather than causing a connection error. The only fatal error is if the client sends more frames to a stream after they've sent a reset (we don't handle this either, there's a skipped h2spec test for it).

Reply:
So there’s a difference between in flight and sent after the abort? We don’t know the second frame is ok flight and it seems a bit odd to special case that no? It would mean we need to drain the socket before aborting any streams?

Reply:
Sorry, I confused things by introducing the reverse scenario. Let me get that out of the way.

Sender abort: It's a connection level protocol error for the sender to send any frames on a stream after they've sent (not received) a reset. This applies to both client and server. There's an h2spec test for this we don't pass at the moment.

But that's not what's happening in this test. In this test the client is sending data frames and the server is sending the reset. The server has no idea when the client will receive the reset and stop sending frames. The reset is a polite, cooperative dismissal. There's nothing in the protocol that says how long the server should filter out frames from the client after a reset. If the client does not stop eventually then the server has no choice but to close the connection. I expect we'll need to introduce a new timeout here that tracks and filters aborted streams for a given time period.